src/utils.js

```diff
@@ -1150,3 +1150,42 @@ export function binarySearch(arr, el, key = (el) => el) {
   }
   return left;
 }
+
+/**
+ * Checks if an object has non-serializable properties.
+ * Non-serializable properties are functions and RegExp objects.
+ *
+ * @param {Object} obj - The object to check.
+ * @param {Set} checkedObjects - A set of properties that have already been checked.
```

Comment:
The `Set` part could be skipped, probably netting some memory wins, but only if we can guarantee that the objects coming in are not circular (`JSON.parse/stringify` would throw anyway if that is the case.)